pkg/cloudprovider/providers/aws/aws_loadbalancer.go

```diff
@@ -308,3 +310,82 @@ func (s *AWSCloud) ensureLoadBalancerInstances(loadBalancerName string, lbInstan
 
 	return nil
 }
+
+func getLoadBalancerHostedZone(service *api.Service) string {
+	if zone, ok := service.Annotations[ServiceAnnotationLoadBalancerCnameZone]; ok {
+		return zone + "."
+	} else {
+		return ""
+	}
+}
+
+func (s *AWSCloud) getLoadBalancerCname(service *api.Service) string {
+	hostedZone := getLoadBalancerHostedZone(service)
+	if hostedZone == "" {
+		return ""
+	}
+
+	return fmt.Sprintf("%v-%v-%v.%v", service.Name, service.Namespace, s.cfg.Global.KubernetesClusterTag, hostedZone)
```

Comment:
So is the idea here that we assign a stable name to the service, and that we then CNAME the actual hostname (`www.mycoolapp.com`) to that stable name?